tests/virt_autotest/virt_autotest_base.pm

```diff
@@ -13,16 +13,26 @@ package virt_autotest_base;
 
 use strict;
 use warnings;
+use Sys::Hostname;
 use File::Basename;
 use base "opensusebasetest";
 use testapi;
 use Data::Dumper;
 use XML::Writer;
 use IO::File;
 use virt_utils;
+use xen;
 
+#This is the base analyzeResult subroutine, which provides simplified core functionalities.
 sub analyzeResult {
-    die "You need to overload analyzeResult in your class";
+    my ($self, $test_status, $test_time) = @_;
+    diag("You need to overload analyzeResult in your class to personalize its usage. This one only provides simplified core functionality");
+    my $result;
+    foreach (keys %xen::guests) {
+        $result->{$_}{status} = $test_status;
+        $result->{$_}{time}   = $test_time;
+    }
+    return $result;
```

Comment:
As you state, this function needs to be general for all class tree from virt_autotest_base, but this implementation is not common/fundamental enough to be shared by all inherited class. SO this is not proper place for it.

Reply:
So what is your ideal proper place ?  You can say it and I won't mind move it to another place. But, for me, this is ideal place for me.